xonsh/base_shell.py

```diff
@@ -15,11 +13,10 @@
 
 
 class TeeOut(object):
-    """Tees stdout into the original sys.stdout and another buffer instance that is 
-    provided.
-    """
+    """Tees stdout into the original sys.stdout and another buffer."""
 
     def __init__(self, buf, *args, **kwargs):
+        _, _ = args, kwargs
```

Comment:
It's not really being silly. It's legitimately pointing out that an argument isn't used which frequently implies either a bug (typically due to a misspelled var) or a potential API bogosity. However, as you point out unused args do have legitimate use cases. Within Google the idiom is to assign unused args to `_` to tell both lint and readers of the code that the args are deliberately being ignored.

Having said that I personally think the `*args, **kwargs` presence in this method are problematic. If the interface that TeeOut is expected to implement changes then the signature of its methods should change. Silently discarding unexpected positional and keyword args is a landmine waiting to hurt someone.